Develop

xcixor opened the pull request:
#### Purpose of the PR
* Creates all the features required by a driver and a passenger to use the application
#### Tasks to be complete
* Show all rides
* Show the details of a ride
* Show the status of a ride
* Show ride history for both the driver and the passenger
* Enable ride requesting
* Enable ride confirmation (reject/accept)
#### Manual testing
* Navigate to the repo's main page via  [this link](https://github.com/xcixor/my-ride)
* Under the repo's name copy the clone's url
* Open a terminal and navigate to where you want the repo to be downloaded
* Type git clone 'the url you copied above' and press enter to download
* Checkout to the branch develop
* Navigate to the UI folder
* Open the index file with your favorite browser
* Start running!
* To preview on mobile in chrome inspect the page and the browser's inspection tool will provide 
   previews for various screens
* Or preview the site [here](https://xcixor.github.io/my-ride/)
#### Relevant pivotal tracker stories 
[user should be able to view available rides ](https://www.pivotaltracker.com/n/projects/2179559)
[User should be able to see the details of a ride and make a request. ](https://www.pivotaltracker.com/n/projects/2179559)
[A driver should be able to see the status of a ride](https://www.pivotaltracker.com/n/projects/2179559)
[A user should be able to see ride history ](https://www.pivotaltracker.com/n/projects/2179559)

xcixor committed:
feat(create landing page

- Create sign up form

Style the sign up form

-Modify universal style to suit all pages

xcixor committed:
feat(navigation)

-Add links for signup page

xcixor committed:
style(code quality)

-remove  commentd code

xcixor committed:
fix(Edit form message)

xcixor committed:
style(Remove unused code)

xcixor committed:
[Fix #158421514: Edit form message]

xcixor committed:
feature(login page)

-Add form to login

-Style the form

[(Finishes) #158421826]

xcixor committed:
Merge branch '158421826-login' of github.com:xcixor/my-ride into 158421826-login

xcixor committed:
feat(Drivers dashboard)

- Add container to hold the driver's rides

-Add a side nav to navigate various aspects of the app

-Add a floating button to create a new ride

xcixor committed:
feat(Drivers dashboard)

-Add style to the side navigation

[Delivers #158423392]

xcixor committed:
feat(Drivers dashboard)

-Add function to open and close side navigation in smaller screens

[Delivers #158423392]

xcixor committed:
Feat(Create ride)

-Add form to collect ride details

xcixor committed:
chore(navigation)

-Add links for existing pages

[Delivers #158423392]

xcixor committed:
Ignore unnecessary images

xcixor committed:
Fix content typos and document heading ommission

xcixor committed:
feat(view all rides)

-Create page to view all rides

xcixor committed:
feat(ride details)

-Add page to display more ride details

-Add button to reques ride

-Add button to add driver as a friend so that user can only respond to offers from their friends

[Finishes #158423079]

xcixor committed:
feat(ride status)

-Create page to view a ride and its reques

-Add functionality to accept or reject a ride

xcixor committed:
feat(passenger ride history)

-Create dashboard to view user's riding history

xcixor committed:
feat(driver histiry)

-Add page to view driver's history

xcixor committed:
fix(navigation)

Add missing links

xcixor committed:
-Add project files

xcixor committed:
Merge branch 'develop' of github.com:xcixor/my-ride into develop

JoshuaOndieki commented on UI/driver_dashboard.html:
```diff
@@ -27,8 +27,8 @@ <h2>Hi Driver welcome to your workspace</h2>
+                <a href="passenger_dashboard.html">Passenger Profile</a>
                 <a href="index.html">Log out</a>
             </div>
             <span style="font-size:30px;cursor:pointer; padding: 20px;" onclick="openNav()">&#9776;</span>
```
Inline CSS :x:

JoshuaOndieki commented on UI/driver_dashboard.html:
```diff
@@ -0,0 +1,136 @@
+                <a href="passenger_dashboard.html">Passenger Profile</a>
+                <a href="index.html">Log out</a>
+            </div>
+            <span style="font-size:30px;cursor:pointer; padding: 20px;" onclick="openNav()">&#9776;</span>
```
Inline CSS :x:

xcixor replied:
let me check it out

xcixor committed:
fix(navbar)

- Add div with mobile menu

-Add styling

-Reference existing js code in index page

[Fixes #158527425]

xcixor committed:
fix(navbar)

-Implement navbar fix for all pages

[Finishes #158527425]

artorious commented:
Hi, As you fix your `CSS` take a look at whats causing the overflow of some elements in small screen.
Earlier versions didn't have this bug. 
 
![screenshot_2018-06-21-17-51-28](https://user-images.githubusercontent.com/15730750/41727341-cc7b6a76-757c-11e8-9604-9537cb71ff6f.png)
![screenshot_2018-06-21-17-51-56](https://user-images.githubusercontent.com/15730750/41727342-ccdf18aa-757c-11e8-92d5-777f735cdf63.png)
![screenshot_2018-06-21-17-52-10](https://user-images.githubusercontent.com/15730750/41727344-cd58fcc4-757c-11e8-931c-6fee945d8d07.png)
![screenshot_2018-06-21-17-52-18](https://user-images.githubusercontent.com/15730750/41727346-cdaa6956-757c-11e8-98b6-da7e1dc8f57b.png)

xcixor committed:
chore(navigation)

-Add missing links

xcixor committed:
-maintain(quality)

-Remove unused code

-Add missing links

-Align misaligned divs

xcixor committed:
fix(mobile view)

-Edit poor presentation in smaller screens

xcixor committed:
-Edit README.md

[Finishes #158429240]

xcixor committed:
Edit Readme to enhance presentation

xcixor committed:
Feat(driver profile

-Add page to view more details about a driver

xcixor committed:
feat(driver public profile)

[Finishes #158558574]

xcixor committed:
feat(ride status)

-Add page where user can view ride status

[Finishes #158558708]

xcixor committed:
feat(user profile)

-Add user profile page

-edit links to direct to proper pages

[Finishes #158560716]

xcixor commented:
#### Purpose of the PR
* Implement LFA feedback
#### Tasks to be complete
* Create user profile page
* Add more details about a ride - number plate
* Public driver profile with contacts
* passenger sees ride status accept or reject
#### Screenshots
##### User profile
![feed1](https://user-images.githubusercontent.com/22756288/41790136-3c4f292c-765a-11e8-8de9-78905348b758.png)

##### Link to view more driver details
![feed2](https://user-images.githubusercontent.com/22756288/41790193-67a2d420-765a-11e8-872a-94fc2f469c56.png)

##### More ride's details
![feed5](https://user-images.githubusercontent.com/22756288/41790899-b95f06c4-765c-11e8-8742-4f0692a631f5.png)


##### Driver's profile
![feed3](https://user-images.githubusercontent.com/22756288/41790212-743eff9c-765a-11e8-9bdc-ef2735cf97f0.png)

##### Ride status
![feed4](https://user-images.githubusercontent.com/22756288/41790297-c23c532a-765a-11e8-83bd-67f96d3254d8.png)

xcixor committed:
feat (ride details)

-Add more details about a ride

JoshuaOndieki commented:
You could have opened a new PR for these changes specifically requested by the LFA.
Anyway, since you have already pushed to this PR and as I can see all requested features have been implemented, you're good to go :+1:
